client/src/common/dataProvider.js

```diff
@@ -1,77 +1,106 @@
+    };
+
+    const url = `${apiUrl}/${resource}/rows?${stringify(query)}`;
+
+    return axios.get(url).then((response) => {
+      const { data } = response.data;
+
+      //manually add an id key
+      const modifiedData = [];
+      for (let i = 0; i < data.length; i++) {
+        const item = data[i];
+        const primaryKey = pkDictionary[resource];
+
+        if (primaryKey !== undefined) {
+          item.id = item[primaryKey];
         }
 
-        response.data.data = modifiedData;
-        return response.data;
-      });
-    },
-
-    create: (resource, params) => {
-      const url = `${apiUrl}/${resource}/rows`;
-
-      return axios.post(url, { fields: params.data }).then((response) => {
-        return { data: { id: response.data.lastInsertRowId, ...params.data } };
-      });
-    },
-
-    getOne: (resource, params) => {
-      const url = `${apiUrl}/${resource}/rows/${params.id}`;
-
-      return axios.get(url).then((response) => {
-        const { data } = response.data;
-        return { data: { id: data[0].GenreId, ...data[0] } };
-      });
-    },
-
-    update: (resource, params) => {
-      const url = `${apiUrl}/${resource}/rows/${params.id}`;
-
-      // remove the id property
-      const { id, ...editData } = params.data;
-
-      return axios.put(url, { fields: editData }).then((response) => {
-        return { data: { id: response.data.lastInsertRowId, ...params.data } };
-      });
-    },
-
-    deleteMany: (resource, params) => {
-      const ids = params.ids.toString();
-      const url = `${apiUrl}/${resource}/rows/${ids}`;
-
-      return axios.delete(url).then((response) => {
-        return { data: { id: response.data.lastInsertRowId, ...params.data } };
-      });
-    },
-  };
-};
+        modifiedData.push(item);
+      }
+
+      response.data.data = modifiedData;
+      return response.data;
+    });
+  },
+
+  create: (resource, params) => {
+    const url = `${apiUrl}/${resource}/rows`;
+
+    return axios.post(url, { fields: params.data }).then((response) => {
+      return { data: { id: response.data.lastInsertRowId, ...params.data } };
+    });
+  },
+
+  getOne: (resource, params) => {
+    const url = `${apiUrl}/${resource}/rows/${params.id}`;
+
+    return axios.get(url).then((response) => {
+      let { data } = response.data;
+      data = data[0];
+
+      const primaryKey = pkDictionary[resource];
+      if (primaryKey !== undefined) {
+        data.id = data[primaryKey];
+      }
+
+      return { data };
+    });
+  },
+
+  getMany: (resource, params) => {
+    const url = `${apiUrl}/${resource}/rows/${params.ids.toString()}`;
+
+    return axios.get(url).then((response) => {
+      const { data } = response.data;
+
+      //manually add an id key
+      const modifiedData = [];
+      for (let i = 0; i < data.length; i++) {
+        const item = data[i];
+        const primaryKey = pkDictionary[resource];
+
+        if (primaryKey !== undefined) {
```

Comment:
@AbegaM - when we insert the new `id`, could remove the existing one please?  Having the unused id results in `ListGuesser` creating an extra column for it.  

That relates to all of the `get` actions, please.

Reply:
ok, i will fix this issue